CONTRIBUTING.md

```diff
@@ -0,0 +1,339 @@
+
+### <a name="4.2.2"></a> 4.2.2 Make Changes
+
+
+Make your changes in your **newly created** branch, the project is organized according to the branch model [Git Flow.](http://git-flow.readthedocs.io/en/latest/)
+
+```console
+    git checkout -b created-branch develop
+```
+
+### <a name="4.2.3"></a> 4.2.3 Commit Your Changes
+A commit, or "revision", is an individual change to a file (or set of files). It's like when you save a file, except with Git, every time you save it creates a unique ID (a.k.a. the "SHA" or "hash") that allows you to keep a record of what changes were made when and by who. Commits usually contain a commit message which is a brief description of what changes were made.
+
+### <a name="4.2.3.1"></a> 4.2.3.1 Rules to Follow
+For commits, we follow the [Conventional Commit](http://conventionalcommits.org/). This leads to **more readable messages** that are easy to follow when looking through the project history. But also, we use the git commit messages to **automatically generate changelogs** from these messages.
+
+### <a name="4.2.3.2"></a> 4.2.3.2 Commit Format
+Each commit message consists of a **header**, a **body** and a **footer**.  The header has a special
+format that includes a **type**, a **scope** and a **description**:
+
+The commit message should be structured as follows:
+
+```
+<type>(<optional scope>): <description>
+<BLANK LINE>
+<optional body>
+<BLANK LINE>
+<optional footer>
+```
+
+### <a name="4.2.3.2.1"></a> 4.2.3.2.1 Header: Writting a `<type>`. 
+Commits must be prefixed with a type, which consists of a verb, **feat, fix, build,** followed by a colon and space.
+
+**Your options:**
+
+* **build**: Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm).
+* **ci**: Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs).
+* **docs**: Documentation only changes.
+* **feat**: A new feature.
+* **fix**: A bug fix.
+* **perf**: A code change that improves performance.
+* **refactor**: A code change that neither fixes a bug or adds a feature.
+* **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc).
+* **test**: Adding missing tests or correcting existing tests.
+
+---
+>**Example for `<type>`:**
+>:point_right:feat:point_left:(parser): add ability to parse arrays
+---
+
+### <a name="4.2.3.2.2"></a> 4.2.3.2.2 Header: Writting the (`<optional scope>`).
+A scope (optional) may be provided to a commit’s type, to provide additional contextual information and is contained in parenthesis.
+
+---
+> **Example for a (`<optional scope>`):**
+> feat:point_right:(parser):point_left:: add ability to parse arrays 
+---
+
+### <a name="4.2.3.2.3"></a> 4.2.3.2.3 Header: Writting a `<description>`
+A description must immediately follow the **`<type>(<optional scope>):`** The description is a short description of the pull request.
```

Comment:
the description is a short description of the commit, not the pull request

Reply:
Thanks!